fog/ledger/enclave/impl/src/oblivious_utils.rs

```diff
@@ -0,0 +1,454 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Contains methods that allow a Fog View Router enclave to combine all of the
+//! Fog View Shard's query responses into one query response that'll be returned
+//! for the client.
+
+use aligned_cmov::{
+    subtle::{Choice, ConstantTimeEq},
+    CMov,
+};
+use alloc::vec::Vec;
+use mc_fog_types::ledger::{KeyImageQuery, KeyImageResult, KeyImageResultCode};
+use mc_watcher_api::TimestampResultCode;
+
+/// The default KeyImageResultCode used when collating the shard responses.
+const DEFAULT_KEY_IMAGE_SEARCH_RESULT_CODE: KeyImageResultCode = KeyImageResultCode::NotSpent;
+
+pub fn collate_shard_key_image_search_results(
+    client_queries: Vec<KeyImageQuery>,
+    shard_key_image_search_results: Vec<KeyImageResult>,
```

Comment:
:wrench: Does this need to take ownership?
```suggestion
    shard_key_image_search_results: &Vec<KeyImageResult>,
```

Reply:
It doesn't, based on line 33, but as a general rule you want `&[Foo]` instead of a `&Vec<Foo>`
```suggestion
    shard_key_image_search_results: &[KeyImageResult],
```